service/controller/resource/tccpnencryption/create.go

```diff
@@ -0,0 +1,69 @@
+package tccpnencryption
+
+import (
+	"context"
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

Comment:
Why do we need a new resource for this? I think we can make the existing one configurable so that we can use it in different controllers.

Reply:
create encryptionkey resource that is used for both tccpn and tcnp